packages/rocketchat-ui/client/views/app/room.js

```diff
@@ -878,6 +878,38 @@ Template.room.events({
 			Meteor.call('sendMessage', msgObject);
 		});
 	},
+	async 'click .pdf-open-in-modal'(event) {
+		event.preventDefault();
+		document.body.style.cursor = 'wait';
+		const pdfBlock = $(event.target).parent();
+		const pdfBlockFreeze = 'pdf-block-freeze';
+		pdfBlock.before(`<div class="${ pdfBlockFreeze }"></div>`);
+		const isElectron = (navigator.userAgent.toLowerCase().indexOf(' electron/') > -1);
+		const pdfLink = this._arguments[1].attachments[0].title_link;
+		const pdfId = `${ pdfLink.split('/')[2] }.pdf`;
+		if (isElectron) {
+			const newWindow = window.open(Meteor.absoluteUrl().replace(/\/$/, '') + pdfLink, pdfId, 'width=1,height=1');
+			newWindow.close();
+		} else {
+			const filePdfBuffer = await new Promise((resolve, reject) =>
+				Meteor.call('getPDFFile', pdfId, (error, result) => {
+					if (error) {
+						return reject(error);
+					}
+					resolve(result);
+				})
+			);
+			const filePdfBlob = new Blob([filePdfBuffer], { type: 'application/pdf' });
+			const dataUrl = URL.createObjectURL(filePdfBlob);
+			modal.open({
+				showConfirmButton: false,
+				text: Blaze.toHTMLWithData(Template.pdfViewer, { dataUrl }),
```

Comment:
Can't you pass the PDF url instead of loading the file in memory to pass a object url?

Reply:
as soon as I pass the direct absolute link in the web browser, the file is downloaded and the document is not displayed in the modal window

![_14](https://user-images.githubusercontent.com/13063572/51528337-18d69600-1e3f-11e9-83ee-859e3b007166.png)

Reply:
You may change [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-file-upload/server/lib/FileUpload.js#L45) to accept parameters when requesting the file, that way you could pass like `?disposition=inline` and replace the `attachment` by `inline`

Reply:
@rodrigok tell me pls, how exactly, i can get into this place from 'click .pdf-open-in-modal'(event) function to overwrite **Content-Disposition** header? i find some places where in my code have simple access [this](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-file-upload/server/config/FileSystem.js#L20) and [this](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-file-upload/server/config/GridFS.js#L159), perhaps maybe there are more places that need processing. I understand what it is not very beautiful, because should be manage headers from one place, but now i can't access to [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-file-upload/server/lib/FileUpload.js#L45).

Moreover, using this workaround, I achieved that I was loading into a modal window in the browser, but not until the end. There are problems with CSP:
![_20](https://user-images.githubusercontent.com/13063572/51598369-0a9f7d00-1f06-11e9-9f7e-7fdd896e3c92.png)

Even when I try to set the title `res.setHeader('Content-Security-Policy', 'default-src \'unsafe-inline\' style-src \'unsafe-inline\'');`

Reply:
The only place you need to change is [this](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-file-upload/server/lib/FileUpload.js#L45) I guess, this is the final part where the file pass before end up at client side, there the content-disposition is defined for every file.

Reply:
well, I will continue to look at how to improve this situation in order to remove the server method that is not needed at all (I have already deleted it)

One more question. Is this a solution that will allow working with pdf from a web browser and an electron application to be on a par with rendering pdf, which is now or instead of it?